web3/exceptions.py

```diff
@@ -192,3 +192,11 @@ class InvalidEventABI(ValueError):
     Raised when the event ABI is invalid.
     """
     pass
+
+
+class SolidityError(ValueError):
+    # Inherits from ValueError for backwards compatibility
+    """
+    Raised on a solidity require/revert
+    """
+    pass
```

Comment:
This seems like a breaking change to the API of expected failures for web3. What was the reasoning for intercepting `TransactionFailed` failures and converting to this error (which also seems poorly named, perhaps `AssertionError(TransactionFailed)` would be better considering Solidity isn't the only smart contract language out there that has a assertion feature)